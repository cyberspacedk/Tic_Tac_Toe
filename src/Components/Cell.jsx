import React from "react";
import styled from "styled-components";

const CellBox = styled.button`
  width: 100px;
  height: 100px;
  line-height: 100px;
  background: #fff;
  border: 1px solid #999;
  font-size: 2.5rem;
  font-weight: bold;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
`;

const Cell = ({ value, click }) =>  <CellBox onClick={click}>{value}</CellBox>; 
export default Cell;
