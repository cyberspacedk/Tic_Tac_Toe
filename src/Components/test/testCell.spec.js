import React from "react";
import { shallow } from "enzyme";
import Cell from "../Cell";
import { shallowToJson } from "enzyme-to-json";

describe("Cell container", () => {
	
  // 1- snapshot
  // 2- check typeof value
  // 3- click on cell 

  const mockClick = jest.fn();
  const props = {
    value: "X",
    onClick: mockClick,
    name: "cell"
  };

  it("snapshot cell", () => {
    const myComponent = shallow(<Cell {...props} />);
    expect(shallowToJson(myComponent)).toMatchSnapshot();
  });

  it("value should be a string", () => {
    const myComponent = shallow(<Cell {...props} />);
    expect(myComponent.text()).toEqual("X");
  });

  it("click on cell", () => {
    const myComponent = shallow(<Cell {...props} />);
    myComponent.find('[name="cell"]').simulate("click");
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
});
