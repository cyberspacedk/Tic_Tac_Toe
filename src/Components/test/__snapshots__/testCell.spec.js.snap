// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell container snapshot cell 1`] = `
<styled.button
  name="cell"
  onClick={[MockFunction]}
>
  X
</styled.button>
`;
